core/src/test/java/org/apache/calcite/materialize/NormalizationTrimFieldTest.java

```diff
@@ -105,4 +107,25 @@
     final String relOptimizedStr = RelOptUtil.toString(relOptimized.get(0).getKey());
     assertThat(isLinux(optimized).matches(relOptimizedStr), is(true));
   }
+
+  @Test void testAggregateRelMdColumnOrigins() {
+    final RelBuilder relBuilder = RelBuilder.create(config().build());
+    final LogicalProject project = (LogicalProject) relBuilder.scan("EMP")
+        .project(relBuilder.field("EMPNO"),
+            relBuilder.field("ENAME"),
+            relBuilder.field("JOB"),
+            relBuilder.field("SAL"),
+            relBuilder.field("DEPTNO")).build();
+    final LogicalAggregate aggregate = (LogicalAggregate) relBuilder.push(project)
+        .aggregate(
+            relBuilder.groupKey(relBuilder.field(1, 0, "DEPTNO")),
+            relBuilder.count(relBuilder.field(1, 0, "SAL")))
+        .build();
+    RelMetadataQuery mq = aggregate.getCluster().getMetadataQuery();
+    final RelColumnOrigin nameColumn = mq.getColumnOrigin(aggregate, 0);
+    assertThat(nameColumn.getOriginTable().getRowType().
+        getFieldNames().get(7).equals("DEPTNO"), is(true));
+    assertThat(nameColumn.getOriginColumnOrdinal() == 7, is(true));
```

Comment:
Please, avoid `assertThat(..., is(true))` pattern because it results in very hard to understand test failures (... expected true got false).

The test failure message should be understandable without looking into the source code and/or JIRA.

Reply:
Thanks, update the code.